packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -592,7 +592,12 @@ export class ForkChoice implements IForkChoice {
    * Returns `true` if the block is known **and** a descendant of the finalized root.
    */
   hasBlockHex(blockRoot: RootHex): boolean {
-    return this.protoArray.hasBlock(blockRoot) && this.isDescendantOfFinalized(blockRoot);
+    const node = this.protoArray.getNode(blockRoot);
+    if (node === undefined) {
+      return false;
+    }
+
+    return this.protoArray.isFinalizedRootOrDescendant(node);
```

Comment:
A bit unrelated to the PR, but can you merge all the if statements in https://github.com/ChainSafe/lodestar/blob/67d0aea6354b577843f516c17bf6da14f547204b/packages/fork-choice/src/protoArray/protoArray.ts#L752

Reply:
I created https://github.com/ChainSafe/lodestar/pull/5908